examples/flutter_gallery/lib/demo/video_demo.dart

```diff
@@ -0,0 +1,290 @@
+    await _controller.dispose();
+  }
+}
+
+class LoopingVideoCard extends StatelessWidget {
+  final VideoPlayerController controller;
+  final String title;
+  final String subtitle;
+
+  const LoopingVideoCard({Key key, this.controller, this.title, this.subtitle})
+      : super(key: key);
+
+  Widget buildInlineVideo() {
+    return new Padding(
+      padding: const EdgeInsets.all(5.0),
+      child: new Center(
+        child: new AspectRatio(
+          aspectRatio: 3 / 2,
+          child: new Hero(
+            tag: title,
+            child: (controller == null)
+                ? new Container()
+                : new VideoPlayer(controller),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget buildFullScreenVideo() {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text(title),
+      ),
+      body: new Center(
+        child: new AspectRatio(
+          aspectRatio: 3 / 2,
+          child: new Hero(
+            tag: title,
+            child: new VideoPlayPause(controller),
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Card(
+      child: new Column(
+        children: <Widget>[
+          new ListTile(
+            title: new Text(title),
+            subtitle: new Text(subtitle),
+          ),
+          new GestureDetector(
+            onTap: () {
+              Navigator.of(context).push(
+                  new MaterialPageRoute<Null>(builder: (BuildContext context) {
+                return buildFullScreenVideo();
+              }));
+            },
+            child: buildInlineVideo(),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class VideoDemo extends StatefulWidget {
+  const VideoDemo({Key key}) : super(key: key);
+
+  static const String routeName = '/video';
+
+  @override
+  VideoDemoState createState() {
+    return new VideoDemoState();
+  }
+}
+
+class VideoPlayPause extends StatefulWidget {
+  final VideoPlayerController controller;
+
+  const VideoPlayPause(this.controller);
+
+  @override
+  State createState() {
+    return new _VideoPlayPauseState();
+  }
+}
+
+class _VideoPlayPauseState extends State<VideoPlayPause> {
+  FadeAnimation imageFadeAnim;
+  VoidCallback listener;
+
+  _VideoPlayPauseState() {
+    listener = () {
+      setState(() {});
+    };
+  }
+
+  VideoPlayerController get controller => widget.controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(listener);
+  }
+
+  @override
+  void deactivate() {
+    controller.removeListener(listener);
+    super.deactivate();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[
+      new GestureDetector(
+        child: new VideoPlayer(controller),
+        onTap: () {
+          if (!controller.value.initialized) {
+            return;
+          }
+          if (controller.value.isPlaying) {
+            imageFadeAnim =
```

Comment:
We can avoid non-Flutterish formatting by inserting some trailing commas in the following lines.